src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClients/FailOverSupportedConfigurationClient.cs

```diff
@@ -0,0 +1,253 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+
+using Azure;
+using Azure.Core;
+using Azure.Data.AppConfiguration;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.ConfigurationClients
+{
+    internal class FailOverSupportedConfigurationClient : IConfigurationClient
+    {
+        private const int HttpStatusCodeRequestThrottled = 429;
+
+        private readonly IEnumerable<(ConfigurationClient, ConfigurationClientState)> _configurationClientAndStates;
```

Comment:
The tuple confuses me a bit. It becomes `Item1` and `Item2`. Any chance we could have a private type to keep the client and any state together?

Reply:
It is actually a conflicting suggestion with @zhenlan suggestion, my first iteration involved a separate type for this, but was shot down because of unnecessary type.

My personal opinion is that adding a type makes the code clean.

Reply:
I've put a private class for now, I can update if @zhenlan has better suggestion.

Reply:
Not that you add more properties, I think it makes sense to have a private type. Please put it in a separate file.

The name is odd. How about `ConfigurationClientWrapper` or something?

Reply:
The use of this class is only limited to `FailOverSupportedClient` aka `FailOverClient`. It would make more sense to have it in this file.
Also, we can't make types private if they are in a separate type. we would have to mark them as internal.
I like your suggestion of renaming it to `ConfigurationClientWrapper`.

Reply:
It's good practice to always put a type in its own file. I don't remember if we have other nested types in the same file, but I may miss something.

Reply:
We have private types in the same file in other repos.
[here](https://msazure.visualstudio.com/Azure%20AppConfig/_git/AppConfigService?path=/src/Microsoft.AppConfig.Service.Authentication.Hmac/HmacTenantContextResolver.cs&version=GBdev&line=31&lineEnd=37&lineStartColumn=9&lineEndColumn=10&lineStyle=plain&_a=contents)